### PR TITLE
fix: tedge write remove hardcoded path

### PR DESCRIPTION
**Pull request by Bravo555**
## Proposed changes

API for spawning a tedge-write process no longer uses a hardcoded `/usr/bin/tedge-write` path.

The path was hardcoded because of the wrong assumption of how `sudo` works: when provided with a command without the full path, e.g. `sudo tedge-write /etc/config1`, `sudo` will internally look for `tedge-write` in $PATH and then compare path with its sudoers entry, so it's not necessary to use full path for binaries used with `sudo`.

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue

- #3073

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [x] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/common/tedge_config/src/sudo.rs
Comment (reubenmiller) [inline crates/common/tedge_config/src/sudo.rs:57]: I'd recommend using the short form option `-n` here (even though it is less readable) because `-n` is more cross compatible, e.g. [freebsd](https://man.freebsd.org/cgi/man.cgi?query=sudo&apropos=0&sektion=0&manpath=FreeBSD+8.1-RELEASE+and+Ports&format=html) only has the `-n` option and not the long form.

**Review by didier-wenzek — APPROVED** [review]
Approved

**Review by gligorisaev — COMMENTED** [review]
Have checked the added Test step and keyword, working as expected

File: crates/common/tedge_config/src/sudo.rs
Comment (Bravo555) [reply crates/common/tedge_config/src/sudo.rs]: fixed

**Review by albinsuresh — APPROVED** [review]
So, we're doing an explicit `which` lookup from the `tedge-agent` process to find `tedge-write` in its customized `$PATH` because `sudo`, by default, ignores the `$PATH` env of the parent process, right?

LGTM and I don't want to block this PR but I'm just wondering if running all sudo commands issued by `tedge-agent` with the `--preserve-env` option would be a better fix, as that'll make sure that not just `tedge-write` but any other process spawned by `tedge-agent`, like the software management plugins as well, would benefit from the same fix. I'm only concerned if it's some sort of "bad security practice" invoking `sudo` preserving the calling process's env.

File: crates/core/tedge_write/src/api.rs
Comment (didier-wenzek) [inline crates/core/tedge_write/src/api.rs:66]: Oops, I have not been fast enough to answer @albinsuresh question before this PR being merged by @Bravo555 !

> I'm just wondering if running all sudo commands issued by `tedge-agent` with the `--preserve-env` option would be a better fix, as that'll make sure that not just `tedge-write` but any other process spawned by `tedge-agent`, like the software management plugins as well, would benefit from the same fix. I'm only concerned if it's some sort of "bad security practice" invoking `sudo` preserving the calling process's env.

One simple way,  to have all the components benefiting from the same fix, is to move the call to `which::which_global()`  *inside* the method `sudo.command()`.

**Pull request merged**
